### PR TITLE
[bitnami/*] Drop support for deprecated cert-manager annotation

### PR DESCRIPTION
**Pull request by pablogalegoc**
**Description of the change**

We are using the `ingress.certManager` parameter to include the `kubernetes.io/tls-acme: "true"` annotation in the Ingress resource to trigger the autogeneration of certificates cert-manager. This annotation comes from [kube-lego](https://github.com/jetstack/kube-lego), which is deprecated. We should instead point the user to add the [currently supported annotations](https://cert-manager.io/docs/usage/ingress/#supported-annotations).

In cases where `ingress.certManager` was used as part of a condition to trigger a behaviour it has been changed by a macro that looks for one of the two strictly required cert-manager annotations: `cert-manager.io/cluster-issuer` or `cert-manager.io/issuer`.

**Benefits**

cert-manager annotations will be just another `ingress.annotations`

**Possible drawbacks**

We are removing the parameter from the `values.yaml` but not from the templates so it is usable in cases where it is still declared. However, the reference in the templates should be removed in the next major.

**Additional info**

  - Relates to #7582
  - This PR won't be merged until there's another PR updating the chart docs (https://github.com/bitnami/charts/pull/7582#pullrequestreview-763076526).

**Checklist** 

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

File: bitnami/ghost/values.yaml [hunk 1]
```diff
@@ -400,9 +407,10 @@ ingress:
   ##   - Relay on Helm to create self-signed certificates by setting `ingress.selfSigned=true`
```
Comment (juan131) [inline bitnami/ghost/values.yaml:407]: Check the previous line (406). My suggestion:
```diff
-  ##   - Relay on cert-manager to create it by setting `ingress.certManager=true`
+  ##   - Relay on cert-manager to create it by setting the corresponding annotations
```

File: bitnami/ghost/values.yaml [hunk 2]
```diff
@@ -388,8 +388,15 @@ ingress:
...
-  ## NOTE: If `ingress.certManager=true`, annotation `kubernetes.io/tls-acme: "true"` will automatically be added
+  ## @param ingress.annotations Additional annotations for the Ingress resource. To enable certificate autogeneration, place here your cert-manager annotations.
+  ## For a full list of possible ingress annotations, please see
+  ## ref: https://github.com/kubernetes/ingress-nginx/blob/master/docs/user-guide/nginx-configuration/annotations.md
+  ## Use this parameter to set the required annotations for cert-manager, see
+  ## ref: https://cert-manager.io/docs/usage/ingress/#supported-annotations
+  ##
+  ## Examples:
+  ## kubernetes.io/ingress.class: nginx
+  ## cert-manager.io/cluster-issuer: cluster-issuer-name
```
Comment (juan131) [inline bitnami/ghost/values.yaml:399]: I personally prefer the examples to include the parameter name:
```suggestion
  ## e.g:
  ## annotations:
  ##   kubernetes.io/ingress.class: nginx
  ##   cert-manager.io/cluster-issuer: cluster-issuer-name
```

File: bitnami/ghost/templates/_helpers.tpl [hunk 3]
```diff
@@ -146,3 +146,13 @@ ghost: database
...
+
+{{/*
+Return true if cert-manager required annotations for TLS signed certificates are set in the Ingress annotations
+Ref: https://cert-manager.io/docs/usage/ingress/#supported-annotations
+*/}}
+{{- define "ghost.ingress.certManagerRequest" -}}
+{{ if or (hasKey . "cert-manager.io/cluster-issuer") (hasKey . "cert-manager.io/issuer") }}
+    {{- true -}}
+{{- end -}}
+{{- end -}}
```
Comment (juan131) [inline bitnami/ghost/templates/_helpers.tpl:158]: Awesome!

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks so mucho for this @pablogalegoc 

Please check my comments! The comments I added to Ghost can be applied to the rest of charts

File: bitnami/ghost/values.yaml (hunk 2 above)
Comment (pablogalegoc) [reply bitnami/ghost/values.yaml]: Done in e0c3a42!

File: bitnami/ghost/values.yaml (hunk 1 above)
Comment (pablogalegoc) [reply bitnami/ghost/values.yaml:408]: Done in bacc9c7!

File: bitnami/ghost/templates/_helpers.tpl (hunk 3 above)
Comment (pablogalegoc) [reply bitnami/ghost/templates/_helpers.tpl:158]: Thanks!

**Review by juan131 — APPROVED** [review]
LGTM

**Pull request merged**